src/config.c

```diff
@@ -688,6 +719,63 @@ void configSetCommand(redisClient *c) {
 
         if (yn == -1) goto badfmt;
         server.aof_rewrite_incremental_fsync = yn;
+#ifdef USE_NVML
+    } else if (!strcasecmp(c->argv[2]->ptr, "aof-use-nvml")){
+        int enable = yesnotoi(o->ptr);
+
+        if (enable == -1) goto badfmt;
+
+        if (server.aof_use_nvml != enable) {
+            if (server.aof_state == REDIS_AOF_OFF) {
+                server.aof_use_nvml = enable;
+            } else if (server.aof_state == REDIS_AOF_ON) {
+                // - change aof-nvml-value
+                // - rewrite aof
+                server.aof_use_nvml = enable;
+                redisLog(REDIS_NOTICE,"Restarting AOF to switch NVML mode");
+                stopAppendOnly();
+                if (startAppendOnly() == REDIS_ERR) {
+                    addReplyError(c,"Unable to turn on AOF. Check server logs.");
+                    return;
+                }
+            } else {
+                // rewrite operation in progress, return error
+                addReplyError(c,
+                    "Unable to change AOF NVML mode, rewrite operation in progress.");
+                return;
+            }
+        }
+    } else if (!strcasecmp(c->argv[2]->ptr, "aof-nvml-direct")){
+        int enable = yesnotoi(o->ptr);
+
+        if (enable == -1) goto badfmt;
+        server.aof_nvml_direct = enable;
+        if (server.aof_use_nvml != REDIS_AOF_OFF
+                && server.aof_use_nvml == 1
+                && server.aof_nvml_direct == 1)
+            flushAppendOnlyFile(0);
+    } else if (!strcasecmp(c->argv[2]->ptr, "aof-nvml-log-size")){
+        // resize allocated nvml log size
+        // 1. if new value is bigger: resize and trigger rewrite
+        // 2. if new value is smaller: respond with error
+        if (getLongLongFromObject(o,&ll) == REDIS_ERR || ll < 0) goto badfmt;
+        if (ll < (server.aof_current_size + PMEMLOG_MIN_POOL)) {
+            addReplyError(c,
+                "Unable to change AOF NVML log size, value to small.");
+            return;
+        } else {
+            server.aof_nvml_log_size = (unsigned)ll;
+            if (server.aof_state == REDIS_AOF_ON && server.aof_use_nvml == 1) {
+                // - rewrite aof
```

Comment:
Yes, it's "by design", starting appendonly mode is automatically triggering rewrite. I'm stopping and starting AOF to:
1) create necessary pools,
2) populate main pool with current runtime data.